banzai_nres/classify.py

```diff
@@ -0,0 +1,73 @@
+from banzai.stages import Stage
+from banzai_nres.frames import NRESObservationFrame
+from banzai_nres import dbs
+import warnings
+from astropy import units
+from astropy.time import Time
+from astropy.coordinates import SkyCoord
+from astroquery.gaia import Gaia
+from astroquery.simbad import Simbad
+simbad = Simbad()
+simbad.add_votable_fields('pmra', 'pmdec', 'fe_h')
+Gaia.ROW_LIMIT = 200
+
+
+def get_initial_guess(ra, dec, pm_ra, pm_dec):
+    """
+
+    :param ra: hour angle
+    :param dec: deg
+    :param pm_ra: mas/yr, Note this needs to include the cos dec term like it is in simbad
+    :param pm_dec: mas/yr
+    :return:
+    """
+
+    # Assume that the equinox and input epoch are both j2000.
+    # Gaia uses an equinox of 2000, but epoch of 2015.5 for the proper motion
+    coordinate = SkyCoord(ra=ra, dec=dec, unit=(units.hourangle, units.deg),
+                          frame='icrs', pm_ra_cosdec=pm_ra * units.mas / units.year,
+                          pm_dec=pm_dec * units.mas / units.year, equinox='j2000',
+                          obstime=Time(2000.0, format='decimalyear'))
+    transformed_coordinate = coordinate.apply_space_motion(new_obstime=Time(2015.5, format='decimalyear'))
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        # 10 arcseconds should be a large enough radius to capture bright objects.
+        results = Gaia.query_object(coordinate=transformed_coordinate, radius=10.0 * units.arcsec)
+
+    # Filter out objects fainter than r=12
+    results = results[results['phot_rp_mean_mag'] < 12.0]
+    # If nothing in Gaia fall back to simbad. This should only be for stars that are brighter than mag = 3
```

Comment:
This is fine, but we should be clear in the documentation that this will also fail if someone tries to observe a star fainter than Rp=12, so users shouldn't expect good results for faint sources.

Reply:
We previously weren't able to acquire a 12th mag target using NRES so I think this is OK.